### PR TITLE
add auth api pkg

### PR DESCRIPTION
**Pull request by upccup**

### CODE REVIEW

**Comment by vitan** [comment]
LGTM

File: src/api/router.go [hunk 1]
```diff
@@ -34,26 +32,9 @@ func (api *Api) ApiRouter() *gin.Engine {
...
-
-		Authorization = chains.Authorization(a)
-		AuthorizeServiceAccess = chains.AuthorizeServiceAccess(a)
-
-		// account mode, Authorization enabled
-		authorizeMiddlewares := make(map[string](func(permissionRequired auth.Permission) gin.HandlerFunc), 0)
-		authorizeMiddlewares["AuthorizeServiceAccess"] = AuthorizeServiceAccess
-		a.RegisterApiForAccount(router, authorizeMiddlewares, chains.Authorization(a), middlewares.ListIntercept())
+		a := &authApi.AccountApi{Config: api.Config, CraneDockerClient: api.Client}
+		a.ApiRegister(router, middlewares.ListIntercept())
```
Comment (cmingxu) [inline src/api/router.go]: ApiRegister looks line a noun， but should be a action indeed.

File: src/api/router.go [hunk 2]
```diff
@@ -34,26 +32,9 @@ func (api *Api) ApiRouter() *gin.Engine {
...
-		}
-
-		if api.Config.AccountAuthenticator == "default" {
-			a.Authenticator = authenticators.NewDefaultAuthenticator()
-		} else if api.Config.AccountAuthenticator == "db" {
-			a.Authenticator = authenticators.NewDBAuthenticator()
-		}
-
-		Authorization = chains.Authorization(a)
-		AuthorizeServiceAccess = chains.AuthorizeServiceAccess(a)
```
Comment (cmingxu) [inline src/api/router.go]: this and above statement override default behaviour of Authorization, AuthorizeServiceAccess base on logic in auth module, what is the plan of giving a way of doing this?

File: src/plugins/auth/middlewares/authorize_service_access.go [hunk 3]
```diff
@@ -7,7 +7,7 @@ import (
 	"github.com/gin-gonic/gin"
 )
 
-func AuthorizeServiceAccess(a *auth.AccountApi) func(permissionRequired auth.Permission) gin.HandlerFunc {
+func AuthorizeServiceAccess() func(permissionRequired auth.Permission) gin.HandlerFunc {
```
Comment (cmingxu) [inline src/plugins/auth/middlewares/authorize_service_access.go]: understood !

File: src/plugins/auth/middlewares/authorization.go [hunk 4]
```diff
@@ -11,7 +11,7 @@ import (
 	"github.com/gin-gonic/gin"
 )
 
-func Authorization(a *auth.AccountApi) gin.HandlerFunc {
+func Authorization(tokenStore auth.TokenStore, authenticator auth.Authenticator) gin.HandlerFunc {
```
Comment (cmingxu) [inline src/plugins/auth/middlewares/authorization.go]: Good to me,  thumb!

**Comment by cmingxu** [comment]
LGTM

**Pull request merged**